uncertainty_baselines/models/segmenter.py

```diff
@@ -0,0 +1,411 @@
+from typing import Any, Callable, Optional, Tuple
+
+import flax.linen as nn
+import jax
+import jax.numpy as jnp
+import ml_collections
+
+Array = Any
+PRNGKey = Any
+Shape = Tuple[int]
+Dtype = Any
+
+
+class IdentityLayer(nn.Module):
+  """Identity layer, convenient for giving a name to an array."""
+
+  @nn.compact
+  def __call__(self, x):
+    return x
+
+
+class AddPositionEmbs(nn.Module):
+  """Adds (optionally learned) positional embeddings to the inputs.
+
+  Attributes:
+    posemb_init: positional embedding initializer.
+  """
+  #TODO(kellybuchanan): check initialization nn.initializers.normal(stddev=0.02) from BERT
+  posemb_init: Callable[[PRNGKey, Shape, Dtype], Array]
+
+  @nn.compact
+  def __call__(self, inputs):
+    """Applies AddPositionEmbs module.
+
+    By default this layer uses a fixed sinusoidal embedding table. If a
+    learned position embedding is desired, pass an initializer to
+    posemb_init.
+
+    Args:
+      inputs: Inputs to the layer.
+
+    Returns:
+      Output tensor with shape `(bs, timesteps, in_dim)`.
+    """
+    # inputs.shape is (batch_size, seq_len, emb_dim).
+    assert inputs.ndim == 3, ('Number of dimensions should be 3,'
+                              ' but it is: %d' % inputs.ndim)
+    pos_emb_shape = (1, inputs.shape[1], inputs.shape[2])
+    pe = self.param('pos_embedding', self.posemb_init, pos_emb_shape)
+    return inputs + pe
+
+
+class MlpBlock(nn.Module):
+  """Transformer MLP / feed-forward block."""
+
+  mlp_dim: int
+  dtype: Dtype = jnp.float32
+  out_dim: Optional[int] = None
+  dropout_rate: float = 0.1
+  kernel_init: Callable[[PRNGKey, Shape, Dtype],
+                        Array] = nn.initializers.xavier_uniform()
+  bias_init: Callable[[PRNGKey, Shape, Dtype],
+                      Array] = nn.initializers.normal(stddev=1e-6)
+  use_bias: bool = True
+  precision: Optional[jax.lax.Precision] = None
+  activation_fn: Callable[[jnp.ndarray], jnp.ndarray] = nn.gelu
+
+  @nn.compact
+  def __call__(self, inputs, *, deterministic):
+    """Applies Transformer MlpBlock module."""
+    actual_out_dim = inputs.shape[-1] if self.out_dim is None else self.out_dim
+    x = nn.Dense(
+        features=self.mlp_dim,
+        dtype=self.dtype,
+        use_bias=self.use_bias,
+        kernel_init=self.kernel_init,
+        bias_init=self.bias_init,
+        precision=self.precision)(  # pytype: disable=wrong-arg-types
+            inputs)
+    x = self.activation_fn(x)
+    x = nn.Dropout(rate=self.dropout_rate)(x, deterministic=deterministic)
+    output = nn.Dense(
+        features=actual_out_dim,
+        dtype=self.dtype,
+        kernel_init=self.kernel_init,
+        bias_init=self.bias_init,
+        use_bias=self.use_bias,
+        precision=self.precision)(  # pytype: disable=wrong-arg-types
+            x)
+    output = nn.Dropout(
+        rate=self.dropout_rate)(
+            output, deterministic=deterministic)
+    return output
+
+
+class Encoder1DBlock(nn.Module):
```

Comment:
Are these modules the same as in vit.py? If so, recommend importing them directly.

Reply:
the encoder in segmenter include an stochastic depth mask which consistently improves the performance of the model. (see Table 2 in https://arxiv.org/pdf/2105.05633.pdf) Would it be best to include a flag for stochastic depth in the vit model or keep the segmenter model as a stand alone implementation?

Reply:
Keeping it standalone here sounds good to me.